File: math/perfect_number/cpp/isPerfect.cpp
#include <iostream>
using namespace std;

bool isPerfect(int);

int main()
{
    int num;
    cin >> num;
    cout << (isPerfect(num) ? "Perfect!" : "No!");

    return 0;
}

bool isPerfect(int num) {
    int sumDiv = 0;
    for (int i = 1; i <= num/2; i++) if (num % i == 0) sumDiv += i;
    return sumDiv == num;
}

File: math/perfect_number/python/is_perfect.py
#!/usr/bin/python3

#Written by Grant Hill

from math import sqrt

def divisorsum(num):
    divsum = 0
    for i in range(1, int(sqrt(num))): #You only have to calculate up to the square rot of the number.
        if num % i == 0:
            divsum += i
            divsum += num // i
    return divsum - num #Don't count the number itself when you're calculating the sum of its divisors. 

def is_perfect(num):
    return num == divisorsum(num)
